src/Symfony/Component/Security/Core/Exception/AccessDeniedException.php

```diff
@@ -57,4 +62,26 @@ public function setSubject($subject)
     {
         $this->subject = $subject;
     }
+
+    /**
+     * Sets an access decision and appends the denied reasons to the exception message.
+     */
+    public function setAccessDecision(AccessDecision $accessDecision)
+    {
+        $this->accessDecision = $accessDecision;
+        if (!$accessDecision->getDeniedVotes()) {
+            return;
+        }
+
+        $reasons = array_map(static function (Vote $vote) {
+            return $vote->getReason();
+        }, $accessDecision->getDeniedVotes());
+
+        $this->message = 'Access Denied: '.rtrim(' '.implode(' ', $reasons));
```

Comment:
What if a custom message was passed in the constructor ?

Reply:
Can you please clarify this comment so I can fix?

Reply:
see the constructor arguments. We can customize the message to use something else than `Access Denied`, but this is lost if this setter is called.